### DISCUSSION TITLE
Ability to save a flow to a file

### DISCUSSION THREAD

**Discussion by pszerszen** [postmanlabs/postman-flows#35] [Ideas]
Would be nice to be able to save a flow to a file, share with teammates and let them import it on their Postmans just like you can do it with Collections.

**Comment by saswatds** [comment]
@pszerszen You can create flows in a Team Workspace and multiple team members will be able to collaborate on a single flow. I recommend using this method instead, for collaborating.

Flows use other resources within a workspace as `references`, and exporting and importing the flow binaries will break those references and it would not work unless all of the referenced resources in the workspace are also exported. Because of the complications export brings with it, and because there is a good alternative to have a team workspace. We might not support exporting anytime soon. 

Will be glad to hear your thoughts on this.

**Reply by dexterlabora** [reply]
Sharing flows with the public to demonstrate common usage patterns is just one example where exporting would be helpful. 

I feel like the flows should be tied to one collection, so all requests would be relative to it.

**Reply by tkulev** [reply]
@saswatds does this means flows are primary for paid users? Because, if you are free plan user, you can not share flows. One suggestion is that the export file (or link) may be backed with all resources needed. This way references can be kept intact.

**Reply by saswatds** [reply]
@tkulev Flows are available to be used by all users. If you intend to share flows with team members, you can add up to 3 people to your team in the [free plan](https://www.postman.com/pricing/). We have not implemented import-export yet because the implementation is still in the beta stage. We want to iterate quickly without worrying about import-export and the additional backward compatibility concerns that serialization brings with it. We are also tracking the request for CLI tool #19 and we will try and solve both these use-cases together in time.

**Reply by spncrgr** [reply]
To add my use case to the mix:
I created a Flow in my personal workspace and now I'd like to share it with the team. It looks like I can't do that, though, and will need to re-create the Flow in my team workspace. So, that's kind of a bummer :)

**Comment by jjruizruiz** [comment]
Hello, first of all, Postman flows is such a great idea.

In my team, we were working on it for a couple of months after it was announced. Now, my company bought a license to include us into a new Workspace for all the people in company.

We cannot import our work from one workspace to the other so we need to start over :(

We would be so pleased if an import-export feature was developed.

Thank you so much for reading me and for creating this amazing tool.

**Reply by saswatds** [reply]
@jjruizruiz Thanks for using flows and all the love. We are soon going to ship the ability to move flows from one workspace to another. Hopefully, that will solve your issue.

**Comment by jrkd** [comment]
Currently requests have a </> button on the right vertical nav that opens up "Code snippet". 

I would love to setup a flow and save it as a code snippet in a particular language. Since the code-gen is already working for requests, i wonder how difficult it would be to incorporate this?

The feature would add heaps of value for developers that can define and test flow integrations in postman, then save a working, tested flow into a piece of code that can be easily incorporated into existing codebases.
